src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PensionAnnexDateStampService.java

```diff
@@ -0,0 +1,89 @@
+
+import static org.springframework.http.MediaType.APPLICATION_PDF_VALUE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.service.DocumentManagementService.CONVERTER;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class PensionAnnexDateStampService {
+
+    private final EvidenceManagementUploadService emUploadService;
+
+    private final EvidenceManagementDownloadService emDownloadService;
+
+    private final GenericDocumentService genericDocumentService;
+
+    static final String FORM_P1_DATE_OF_ORDER_TEXTBOX_NAME = "Date the court made/varied/discharged an order";
+
+    public CaseDocument appendApprovedDateToDocument(CaseDocument document,
+                                                     String authToken,
+                                                     LocalDate approvalDate,
+                                                     String caseId) throws Exception {
+        log.info("Adding date stamp to Pension Sharing Annex : {}", document);
+        Optional<LocalDate> optionalApprovalDate = Optional.ofNullable(approvalDate);
+        if (optionalApprovalDate.isPresent()) {
+            byte[] docInBytes = emDownloadService.download(document.getDocumentBinaryUrl(), authToken);
+            byte[] approvedDoc = appendApprovedDateToDocument(docInBytes, approvalDate);
+            MultipartFile multipartFile =
+                FinremMultipartFile.builder().name(document.getDocumentFilename()).content(approvedDoc)
+                    .contentType(APPLICATION_PDF_VALUE).build();
+            List<FileUploadResponse> uploadResponse =
+                emUploadService.upload(Collections.singletonList(multipartFile), caseId, authToken);
+            FileUploadResponse fileSaved = Optional.of(uploadResponse.get(0))
+                .filter(response -> response.getStatus() == HttpStatus.OK)
+                .orElseThrow(() -> new DocumentStorageException("Failed to store document"));
+            Document dateStampedDocument = CONVERTER.apply(fileSaved);
+            return genericDocumentService.toCaseDocument(dateStampedDocument);
+        } else {
+            throw new StampDocumentException("Missing or Invalid Approved Date of Order.");
+        }
+    }
+
+    private byte[] appendApprovedDateToDocument(byte[] inputDocInBytes, LocalDate approvalDate) throws StampDocumentException, IOException {
+        PDDocument doc = Loader.loadPDF(inputDocInBytes);
+        doc.setAllSecurityToBeRemoved(true);
+        Optional<PDAcroForm> acroForm = Optional.ofNullable(doc.getDocumentCatalog().getAcroForm());
+        if (acroForm.isPresent() && acroForm.get().getField(FORM_P1_DATE_OF_ORDER_TEXTBOX_NAME) != null
+            && (acroForm.get().getField(FORM_P1_DATE_OF_ORDER_TEXTBOX_NAME) instanceof PDTextField)) {
+            PDField field = acroForm.get().getField(FORM_P1_DATE_OF_ORDER_TEXTBOX_NAME);
+            PDTextField textBox = (PDTextField) field;
+            textBox.setDefaultAppearance("/Helv 12 Tf 0 g");
```

Comment:
Are we sure we want to add this font in? Can we not used any embedded font, we saw on the D11 that this font was marked as not embedded on the pre-flight tool.